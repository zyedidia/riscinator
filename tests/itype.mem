03f00093
0430f113
04016193
0401c213
fff06293
00002313
00a02393
00400413
00142023
fe242e23
00342223
00042483
ffc42503
00442583
