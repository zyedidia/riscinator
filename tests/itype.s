.section ".text.boot"

.globl _start
_start:
addi x1,   x0,    63  # x1 = 63
andi x2,   x1,    67  # x2 = 3
ori  x3,   x2,    64  # x3 = 67
xori x4,   x3,    64  # x4 = 3
ori  x5,   x0,    -1  # x5 = -1
slti x6,   x0,    0   # x6 = 0
slti x7,   x0,    10  # x7 = 1
slli x8,   x7,    20  # x8 = 0x100000
addi x8,   x8,    4   # x8 = 0x100004
sw   x1,   0(x8)      # 0x4 = 63
sw   x2,   -4(x8)     # 0x0 = 3
sw   x3,   4(x8)      # 0x8 = 67
lw   x9,   0(x8)      # x9 = 63
lw   x10,  -4(x8)     # x10 = 3
lw   x11,  4(x8)      # x11 = 67
done: j done
