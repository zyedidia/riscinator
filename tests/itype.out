x1 = 63
x2 = 3
x3 = 67
x4 = 3
x5 = -1
x6 = 0
x7 = 1
x8 = 4
x9 = 63
x10 = 3
x11 = 67

[0x0] = 3
[0x4] = 63
[0x8] = 67
